storage/options.go

```diff
@@ -174,49 +177,49 @@ func (o *options) EncodingTszPooled() Options {
 	readerIteratorPool := encoding.NewReaderIteratorPool(0)
 	multiReaderIteratorPool := encoding.NewMultiReaderIteratorPool(0)
 
-	encodingOpts := tsz.NewOptions().
+	encodingOpts := encoding.NewOptions().
 		BytesPool(bytesPool).
 		EncoderPool(encoderPool).
 		ReaderIteratorPool(readerIteratorPool).
 		SegmentReaderPool(segmentReaderPool)
 
 	// initialize encoder pool
 	encoderPool.Init(func() encoding.Encoder {
-		return tsz.NewEncoder(timeZero, nil, encodingOpts)
+		return m3tsz.NewEncoder(timeZero, nil, defaultIntOptimizationEnabled, encodingOpts)
```

Comment:
yeah if we make `intOptimizationEnabled` part of the encoding options, you won't need to pass it in explicitly